pandas/tests/indexes/test_base.py

```diff
@@ -1615,6 +1632,122 @@ def test_string_index_repr(self):
 
                 self.assertEqual(coerce(idx), expected)
 
+    def test_mixed_int_index(self):
+        # Mostly the tests from common.py from which mixedIndex was excluded.
+        # The results differ in py2 and py3 because ints and strings
+        # are uncomparable in py3 (GH 13514)
+        ind = Index([0, 'a', 1, 'b', 2, 'c'])    # == self.mixedIndex
+
+        # test_order
+        # 9816 deprecated
+        if PY3:
+            with tm.assertRaisesRegexp(TypeError, "unorderable types"):
+                with tm.assert_produces_warning(FutureWarning):
+                    ind.order()
+        else:
+            with tm.assert_produces_warning(FutureWarning):
+                ind.order()
+
+        # test_argsort
+        if PY3:
+            with tm.assertRaisesRegexp(TypeError, "unorderable types"):
+                result = ind.argsort()
+        else:
+            result = ind.argsort()
+            expected = np.array(ind).argsort()
+            tm.assert_numpy_array_equal(result, expected, check_dtype=False)
+
+        # test_numpy_argsort
+        if PY3:
+            with tm.assertRaisesRegexp(TypeError, "unorderable types"):
+                result = np.argsort(ind)
+        else:
+            result = np.argsort(ind)
+            expected = ind.argsort()
+            tm.assert_numpy_array_equal(result, expected)
+
+        # test_copy_name
+        # Check that "name" argument passed at initialization is honoured
+        # GH12309
+        first = ind.__class__(ind, copy=True, name='mario')
+        second = first.__class__(first, copy=False)
+
+        # Even though "copy=False", we want a new object.
+        self.assertIsNot(first, second)
+        # Not using tm.assert_index_equal() since names differ:
+        self.assertTrue(ind.equals(first))
+
+        self.assertEqual(first.name, 'mario')
+        self.assertEqual(second.name, 'mario')
+
+        s1 = Series(2, index=first)
+        s2 = Series(3, index=second[:-1])
+        if PY3:
+            with tm.assert_produces_warning(RuntimeWarning):
+                # unorderable types
+                s3 = s1 * s2
+        else:
+            s3 = s1 * s2
+        self.assertEqual(s3.index.name, 'mario')
+
+        # test_union_base
+        first = ind[3:]
+        second = ind[:5]
+
+        if PY3:
+            with tm.assert_produces_warning(RuntimeWarning):
+                # unorderable types
+                result = first.union(second)
+                expected = Index(['b', 2, 'c', 0, 'a', 1])
+                self.assert_index_equal(result, expected)
+        else:
+            result = first.union(second)
+            expected = Index(['b', 2, 'c', 0, 'a', 1])
```

Comment:
(Another inconsistency in `union`: in python2, sortedness of a mixed-int result Index is a bit unpredictable.)